### PR TITLE
[Documentation] Cookbook - How to add a custom Model?

### PR DESCRIPTION
**Pull request by TheMadeleine**
| Q               | A
| --------------- | ---
| Doc fix?        | no
| New docs?    | yes
| BC breaks?      | no
| Related tickets | partially #5275 
| License         | MIT

### CODE REVIEW

File: docs/cookbook/custom-model.rst [hunk 1]
```diff
@@ -0,0 +1,158 @@
+How to add a custom model?
+==========================
+
+In some cases you may be needing to add new models to your application in order to cover unique business needs.
+The proccess of extending Sylius with new entities is simple and intuitive.
+
+As an example we will take a **Supplier entity**, which may be really useful for th shop maintenance.
```
Comment (isometriks) [inline docs/cookbook/custom-model.rst]: th => the? Also still works if you just remove it entirely.

File: docs/cookbook/custom-model.rst [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+The generator will ask you for the entity name and fields. See how it should look like to match our assumptions.
+
+.. image:: ../_images/generating_entity.png
+    :align: center
+
+3. Update the database using migrations
+---------------------------------------
+
+Assuming that you have the databease updated to the state before adding the new entity you should run:
```
Comment (gdecorbiac) [inline docs/cookbook/custom-model.rst]: typo error : 'databease' should be 'database'

Comment (gdecorbiac) [inline docs/cookbook/custom-model.rst]: And I'm not a native English speaker but I would turn this phrase into :
`Assuming that your database was up-to-date before adding the new entity, run:`

If a native English speaker could correct me ;-)

File: docs/cookbook/custom-model.rst [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+                all:
+                    subheader: app.ui.supplier
+                index:
+                    icon: 'file image outline'
+        type: sylius.resource
+
+.. code-block:: yaml
+
+    # app/config/routing/admin/admin.yml
+    sylius_admin_carousel_item:
```
Comment (GSadee) [inline docs/cookbook/custom-model.rst]: I guess it should be `app_admin_supplier`

File: docs/cookbook/custom-model.rst [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+.. code-block:: bash
+
+    $ php app/console debug:container | grep supplier
+
+The output should be:
+
+.. image:: ../_images/container_debug_supplier.png
+    :align: center
+
+6. Use Sylius API to create new resource
```
Comment (pjedrzejewski) [inline docs/cookbook/custom-model.rst]: We should mention that this step is optional. (using the API)

File: docs/cookbook/custom-model.rst [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+
+Use the ``access_token`` in the request that will create a new Supplier.
+
+.. code-block:: bash
+
+    $ curl -i -X POST -H "Content-Type: application/json" -H "Authorization: Bearer DEF" -d '{"name": "Example", "description": "Lorem ipsum", "enabled": true}' http://localhost:8000/api/suppliers/
+
+.. tip::
+
+    Read more about authorizing in API :doc:`here </api/authorization>`. Different requests to Sylius API are described in `the Lionframe docs <http://lakion.com/lionframe>`_.
```
Comment (pjedrzejewski) [inline docs/cookbook/custom-model.rst]: I think we should add examples of getting a list of suppliers and partial update of a supplier.

File: docs/cookbook/custom-model.rst [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+
+    # app/config/routing/admin/admin.yml
+    sylius_admin_carousel_item:
+        resource: 'supplier.yml'
+
+.. code-block:: yaml
+
+    # app/config/routing.yml
+    app_admin:
+        resource: 'routing/admin/admin.yml'
```
Comment (pjedrzejewski) [inline docs/cookbook/custom-model.rst]: The main admin routing file should be just ``routing/admin.yml``.

File: docs/cookbook/custom-model.rst [hunk 7]
```diff
@@ -0,0 +1,247 @@
...
+.. tip::
+
+    See :doc:`how to add links to your new entity administration in the administration menu </customization/menu>`.
+
+9. Check the admin panel for your changes
+-----------------------------------------
+
+.. tip::
+
+    To see what you can do with your new entity access the ``http://localhost:8000/admin/suppliers`` url.
```
Comment (pjedrzejewski) [inline docs/cookbook/custom-model.rst]: ``/`` missing at the end.

File: docs/cookbook/custom-model.rst (hunk 5 above)
Comment (TheMadeleine) [inline docs/cookbook/custom-model.rst]: Well I wasn't sure. API is not the main issue of this cookbook.

File: docs/cookbook/custom-model.rst [hunk 8]
```diff
@@ -142,17 +146,109 @@ Use the ``access_token`` in the request that will create a new Supplier.
...
+.. code-block:: yaml
+
+    # app/config/grids/admin/supplier.yml
+    sylius_grid:
+        grids:
+            app_admin_supplier:
+                driver:
+                    name: doctrine/orm
+                    options:
+                        class: AppBundle/Entity/Supplier
```
Comment (gdecorbiac) [inline docs/cookbook/custom-model.rst]: it should be backslashes instead of slashes -> AppBundle\Entity\Supplier

**Comment by pjedrzejewski** [comment]
Nice work Magda, thank you! :)

**Pull request merged**